tests/integration.rs

```diff
@@ -17,76 +17,72 @@
-use secret_vault::msg::{CountResponse, HandleMsg, InitMsg, QueryMsg};
+use secret_vault::contract::{handle, init};
+use secret_vault::msg::{HandleMsg, InitMsg};
 
 // This line will test the output of cargo wasm
-static WASM: &[u8] = include_bytes!("../target/wasm32-unknown-unknown/release/secret_vault.wasm");
+//static WASM: &[u8] = include_bytes!("../target/wasm32-unknown-unknown/release/secret_vault.wasm");
 // You can uncomment this line instead to test productionified build from rust-optimizer
 // static WASM: &[u8] = include_bytes!("../contract.wasm");
 
 #[test]
-fn proper_initialization() {
-    let mut deps = mock_instance(WASM, &[]);
-
-    let msg = InitMsg::Init { count: 17 };
+fn init_sign() {
+    let mut deps = mock_dependencies(20, &[]);
+    let seed_phrase = "GX4CRn8cLOtaxWAduRKr";
+    let key_seed = "uI67K87zFSU3WKlXtiYK";
+    let passphrase = "mmSnja3rrA8J9NWImqOz";
+    let data = "674de7af309b691bc121e6f40a6c76a513883925f5358838fc5943db720a6e78";
+
+    let msg = InitMsg::Init {
+        seed_phrase: seed_phrase.to_string(),
+    };
     let env = mock_env("creator", &coins(1000, "earth"));
+    let res = init(&mut deps, env, msg);
+    println!("{:?}", res.unwrap());
 
-    // we can just call .unwrap() to assert this was a success
-    let res: InitResponse = init(&mut deps, env, msg).unwrap();
-    assert_eq!(res.messages.len(), 0);
-
-    // it worked, let's query the state
-    let res = query(&mut deps, QueryMsg::GetCount {}).unwrap();
-    let value: CountResponse = from_binary(&res).unwrap();
-    assert_eq!(value.count, 17);
-}
-
-#[test]
-fn increment() {
-    let mut deps = mock_instance(WASM, &coins(2, "token"));
-
-    let msg = InitMsg { count: 17 };
-    let env = mock_env("creator", &coins(2, "token"));
-    let _res: InitResponse = init(&mut deps, env, msg).unwrap();
-
-    // beneficiary can release it
-    let env = mock_env("anyone", &coins(2, "token"));
-    let msg = HandleMsg::Increment {};
-    let _res: HandleResponse = handle(&mut deps, env, msg).unwrap();
-
-    // should increase counter by 1
-    let res = query(&mut deps, QueryMsg::GetCount {}).unwrap();
-    let value: CountResponse = from_binary(&res).unwrap();
-    assert_eq!(value.count, 18);
+    let msg = HandleMsg::NewKey {
+        key_seed: key_seed.to_string(),
+        passphrase: passphrase.to_string(),
+    };
+    let env = mock_env("creator", &coins(1000, "earth"));
+    let res = handle(&mut deps, env, msg);
+    let new_key_res = res.unwrap();
+    let api_key = match get_log_attribute(&new_key_res, "api_key".to_string()) {
+        Some(d) => d,
+        None => panic!("No api key detected"),
+    };
+    let key_id = match get_log_attribute(&new_key_res, "key_id".to_string()) {
+        Some(d) => d,
+        None => panic!("No key id detected"),
+    };
+
+    let msg = HandleMsg::Sign {
+        passphrase: passphrase.to_string(),
+        api_key: api_key,
+        key_id: key_id,
+        data: data.to_string(),
+    };
```

Comment:
in struct initialization, the pattern `foo: foo` can be abreviated to just `foo`:
```rust
let msg = HandleMsg::Sign {
    passphrase: passphrase.to_string(),
    data: data.to_string(),
    api_key,
    key_id,
};
```